src/engraving/libmscore/beam.cpp

```diff
@@ -941,8 +941,15 @@ void Beam::extendStems(const std::vector<ChordRest*>& chordRests, PointF start,
         }
         Chord* chord = toChord(chordRest);
         PointF anchor = chordBeamAnchor(chord);
-        qreal proportionAlongX = (anchor.x() - start.x()) / (end.x() - start.x());
-        qreal desiredY = proportionAlongX * (end.y() - start.y()) + start.y();
+        qreal desiredY;
+        if (end.x() > start.x()) {
+            qreal proportionAlongX = (anchor.x() - start.x()) / (end.x() - start.x());
+            desiredY = proportionAlongX * (end.y() - start.y()) + start.y();
+        } else {
+            // This is needed when beam layout is called before horizontal spacing, because
+            // in that case end.x() may be equal to start.x(). See layoutMeasure::getNextMeasure.
```

Comment:
I don't see a different solution at the moment: the method has to work also for `end.x() == start.x()` so that check is necessary to avoid dividing by zero. I'll cancel the comment, though, it is in fact it's unnecessary